manifests/server/root_password.pp

```diff
@@ -2,12 +2,27 @@
 class mysql::server::root_password {
 
   $options = $mysql::server::options
+  $secret_file = $mysql::server::install_secret_file
+
+  # New installations of MySQL will configure a default random password for the root user
+  # with an expiration. No actions can be performed until this password is changed. The
+  # below exec will remove this default password. If the user has supplied a root
+  # password it will be set further down with the mysql_user resource.
+  $ch_pass_command = join([
+    "/usr/bin/mysqladmin -u root --password=\$(awk '/password/{print \$NF}' ${secret_file}) password ''",
```

Comment:
this seems like a really awkward way to get at the password.
i'm not sure /usr/bin/mysqladmin is what we want here. on other systems it might be in different locations, it might be better to work with path => instead